tests/azure_iot_provisioning_client_ut.c

```diff
@@ -583,8 +605,14 @@ static void testAzureIoTProvisioningClient_Register_RegistrationFailure( void **
                       eAzureIoTProvisioningPending );
 
     /* Read response */
-    xResult = AzureIoTProvisioningClient_Register( &xTestProvisioningClient, 0 );
-    assert_int_equal( xResult, eAzureIoTProvisioningServerError );
+    assert_int_equal( AzureIoTProvisioningClient_Register( &xTestProvisioningClient, 0 ),
+                      eAzureIoTProvisioningServerError );
+
+    /* Read extended code */
+    assert_int_equal( AzureIoTProvisioningClient_GetExtendedCode( &xTestProvisioningClient,
+                                                                  &ulCode ),
+                      eAzureIoTProvisioningSuccess );
+    assert_int_equal( ulExtendedCode, ulCode );
```

Comment:
should this be something like
```suggestion
    assert_int_equal( ulExpectedExtendedCode, ulExtendedCode );
```